### PR TITLE
Allowing to set acronyms via acronyms.json

### PR DESCRIPTION
**Pull request by paganotoni**
@markbates this PR allows to set acronyms int the `inflections.json` file or `INFLECT_PATH`.
It uses acronyms in the Ident Parts identification.

### CODE REVIEW

**Comment by markbates** [comment]
@paganotoni @stanislas-m I've been thinking about this one, and I don't like the approach of create an "ad-hoc" api instead of the `inflections.json` file. I have a few suggestions, but none of them is "wow":

1. create an `acronyms.json` concept. this would require plumbing in other packages to pick up and "handle". I like this approach, but it means a decent amount of work, including updated `inflect` as well.
2. re-write existing `inflections.json` files to use a more `structured` approach. Requires back-porting to `inflect` as well and being able to handle two different styles of JSON. it's not overly complex, but it needs to be thought about.

That's the best I've got. I'm open to discussion.

**Comment by paganotoni** [comment]
@markbates I think we could go for the first one and in case we need something extra we could consider building a more structured way of doing it.

One thing I don't have clear is why do we need to port to `inflect`?

**Comment by paganotoni** [comment]
@markbates updated this PR implementing the first of your suggestions

**Comment by paganotoni** [comment]
When you have the chance let me know if this is enough :)

**Comment by mclark4386** [comment]
Obviously, @markbates should be the one to merge this, but looks good to me^^

File: custom_data.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+//CustomDataParser are functions that parse data like acronyms or
+//plurals in the shape of a io.Reader it receives.
+type CustomDataParser func(io.Reader) error
+
+func loadCustomData(defaultFile, env, readErrorMessage string, parser CustomDataParser) {
+	pwd, _ := os.Getwd()
+	path := filepath.Join(pwd, defaultFile)
+
+	if p := os.Getenv(env); p != "" {
```
Comment (markbates) [inline custom_data.go]: Envy

**Comment by paganotoni** [comment]
@markbates moved it to use envy

File: custom_data.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+func LoadAcronyms(r io.Reader) error {
+	m := []string{}
+	err := json.NewDecoder(r).Decode(&m)
+
+	if err != nil {
+		return fmt.Errorf("could not decode acronyms JSON from reader: %s", err)
+	}
+
+	for _, acronym := range m {
+		baseAcronyms[acronym] = true
```
Comment (markbates) [inline custom_data.go]: we probably need a mutes near here

**Pull request merged**